validator/src/journal/publisher.rs

```diff
@@ -479,6 +479,20 @@ impl SyncBlockPublisher {
     }
 
     fn cancel_block(&self, state: &mut BlockPublisherState, unref_block: bool) {
+        let invalid_batches = if let Some(ref cb) = state.candidate_block {
+            cb.invalid_batches.clone()
```

Comment:
This could probably be done with out the clone, give that you don't need to pass ownership of those invalid batches.

Reply:
You're probably right. It was a tiny bit hairy with state owning both the candidate_block and the pending_batches. This was a sure-fire way to get it done.